### ISSUE TITLE
Installing through npm fails on windows

### ISSUE THREAD

**Issue by draperjames** [n-riesco/jp-coffeescript#7]
# Problem: installing through npm on windows fails.
My system specs;
```
Current conda install:

               platform : win-64
          conda version : 4.3.5
       conda is private : False
      conda-env version : 4.3.5
    conda-build version : 1.21.3
         python version : 3.5.2.final.0
       requests version : 2.12.4
       root environment : C:\Anaconda3  (writable)
    default environment : C:\Anaconda3
       envs directories : C:\Anaconda3\envs
          package cache : C:\Anaconda3\pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/win-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/win-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/win-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/msys2/win-64
                          https://repo.continuum.io/pkgs/msys2/noarch
            config file : None
           offline mode : False
             user-agent : conda/4.3.5 requests/2.12.4 CPython/3.5.2 Windows/7 Windows/6.1.7601
```
```
{ npm: '3.8.3',
  ares: '1.10.1-DEV',
  http_parser: '2.5.2',
  modules: '46',
  node: '4.4.1',
  openssl: '1.0.2g',
  uv: '1.8.0',
  v8: '4.5.103.35',
  zlib: '1.2.8' }
```
When I run:

```
npm install -g jp-coffeescript
```
I get the following error:

```
C:\Users\Lenovo>npm install -g jp-coffeescript
npm WARN deprecated node-uuid@1.4.7: use uuid module instead
C:\Anaconda3\Library\bin\jp-coffee -> C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\bin\jp-coffeescript.js

> zmq@2.15.3 install C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\node_modules\zmq
> node-gyp rebuild


C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\node_modules\zmq>if not defined npm_config_node_gyp (node "C:\Anaconda3\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
gyp ERR! configure error
gyp ERR! stack Error: Python executable "C:\Anaconda3\python.EXE" is v3.5.2, which is not supported by gyp.
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & < 3.0.0.
gyp ERR! stack     at failPythonVersion (C:\Anaconda3\node_modules\npm\node_modules\node-gyp\lib\configure.js:406:14)
gyp ERR! stack     at C:\Anaconda3\node_modules\npm\node_modules\node-gyp\lib\configure.js:395:9
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:204:7)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at maybeClose (internal/child_process.js:827:16)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Anaconda3\\Library\\bin\\node.exe" "C:\\Anaconda3\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\node_modules\zmq
gyp ERR! node -v v4.4.1
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok
C:\Anaconda3\Library\bin
`-- (empty)

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Anaconda3\\Library\\bin\\node.exe" "C:\\Anaconda3\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "jp-coffeescript"
npm ERR! node v4.4.1
npm ERR! npm  v3.8.3
npm ERR! code ELIFECYCLE

npm ERR! zmq@2.15.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the zmq@2.15.3 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the zmq package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs zmq
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls zmq
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Lenovo\npm-debug.log
npm ERR! code 1
```

Please let me know if you need any more information.

**Comment by slavaGanzin** [comment]
```
gyp ERR! stack Error: Python executable "C:\Anaconda3\python.EXE" is v3.5.2, which is not supported by gyp.
gyp ERR! stack You can pass the --python switch to point to Python >= v2.5.0 & < 3.0.0.
```

node-gyp works only with python2

**Comment by draperjames** [comment]
Alright I installed node-gyp according to [node-gyp installation guide](https://github.com/nodejs/node-gyp#installation).

So now when I run:
```
npm install -g jp-coffeescript
```

I get these errors:
```
C:\Users\Lenovo>npm install -g jp-coffeescript
npm WARN deprecated node-uuid@1.4.7: use uuid module instead
C:\Anaconda3\Library\bin\jp-coffee -> C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\bin\jp-coffeescript.js

> zmq@2.15.3 install C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\node_modules\zmq
> node-gyp rebuild


C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\node_modules\zmq>if not defined npm_config_node_gyp (node "C:\Anaconda3\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  binding.cc
..\binding.cc(673): warning C4244: 'argument': conversion from 'int64_t' to 'const int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffe
escript\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(674): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescrip
t\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(675): warning C4244: 'argument': conversion from 'int64_t' to 'const int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffe
escript\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(676): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescrip
t\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(677): warning C4244: 'argument': conversion from 'int64_t' to 'const int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffe
escript\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(678): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescrip
t\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(679): warning C4244: 'argument': conversion from 'int64_t' to 'const int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffe
escript\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(680): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescrip
t\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(681): warning C4244: 'argument': conversion from 'int64_t' to 'const int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffe
escript\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(682): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescrip
t\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(696): warning C4244: 'argument': conversion from 'int64_t' to 'const int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffe
escript\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(697): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescrip
t\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(698): warning C4244: 'argument': conversion from 'int64_t' to 'const int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffe
escript\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(699): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescrip
t\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(700): warning C4244: 'argument': conversion from 'int64_t' to 'const int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffe
escript\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(701): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescrip
t\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(702): warning C4244: 'argument': conversion from 'int64_t' to 'const int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffe
escript\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(703): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescrip
t\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(704): warning C4244: 'argument': conversion from 'int64_t' to 'const int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffe
escript\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(705): warning C4244: 'argument': conversion from 'int64_t' to 'int', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescrip
t\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(1140): warning C4267: 'argument': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffee
script\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(1285): warning C4267: 'argument': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffee
script\node_modules\zmq\build\zmq.vcxproj]
..\binding.cc(1286): warning C4267: 'argument': conversion from 'size_t' to 'uint32_t', possible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffee
script\node_modules\zmq\build\zmq.vcxproj]
c:\anaconda3\library\bin\node_modules\jp-coffeescript\node_modules\nan\nan_new.h(208): warning C4244: 'argument': conversion from 'int64_t' to 'double', possi
ble loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\node_modules\zmq\build\zmq.vcxproj]
  ..\binding.cc(624): note: see reference to function template instantiation 'v8::Local<v8::Number> Nan::New<v8::Number,T>(A0)' being compiled
          with
          [
              T=int64_t,
              A0=int64_t
          ]
  ..\binding.cc(678): note: see reference to function template instantiation 'v8::Local<v8::Value> zmq::Socket::GetSockOpt<int64_t>(int)' being compiled
c:\anaconda3\library\bin\node_modules\jp-coffeescript\node_modules\nan\nan_new.h(208): warning C4244: 'argument': conversion from 'uint64_t' to 'double', poss
ible loss of data [C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\node_modules\zmq\build\zmq.vcxproj]
  ..\binding.cc(624): note: see reference to function template instantiation 'v8::Local<v8::Number> Nan::New<v8::Number,T>(A0)' being compiled
          with
          [
              T=uint64_t,
              A0=uint64_t
          ]
  ..\binding.cc(680): note: see reference to function template instantiation 'v8::Local<v8::Value> zmq::Socket::GetSockOpt<uint64_t>(int)' being compiled
  win_delay_load_hook.c
C:\Anaconda3\node_modules\npm\node_modules\node-gyp\src\win_delay_load_hook.c(34): error C2373: '__pfnDliNotifyHook2': redefinition; different type modifiers
[C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\node_modules\zmq\build\zmq.vcxproj]
  C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\delayimp.h(134): note: see declaration of '__pfnDliNotifyHook2'
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\MSBuild\14.0\bin\msbuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (C:\Anaconda3\node_modules\npm\node_modules\node-gyp\lib\build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:87:13)
gyp ERR! stack     at ChildProcess.emit (events.js:172:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
gyp ERR! System Windows_NT 6.1.7601
gyp ERR! command "C:\\Anaconda3\\Library\\bin\\node.exe" "C:\\Anaconda3\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Anaconda3\Library\bin\node_modules\jp-coffeescript\node_modules\zmq
gyp ERR! node -v v4.4.1
gyp ERR! node-gyp -v v3.3.1
gyp ERR! not ok
C:\Anaconda3\Library\bin
`-- (empty)

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Anaconda3\\Library\\bin\\node.exe" "C:\\Anaconda3\\node_modules\\npm\\bin\\npm-cli.js" "install" "-g" "jp-coffeescript"
npm ERR! node v4.4.1
npm ERR! npm  v3.8.3
npm ERR! code ELIFECYCLE

npm ERR! zmq@2.15.3 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the zmq@2.15.3 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the zmq package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs zmq
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls zmq
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\Lenovo\npm-debug.log
npm ERR! code 1
```

**Comment by AurielleP** [comment]
I get a similar error with mac os - ive tried it a million times, really frusterating

**Comment by n-riesco** [comment]
@AurielleP Could you open a new issue and post the error message you get?

**Comment by n-riesco** [comment]
@slavaGanzin @draperjames @AurielleP I've just published version 1.0.0 of `jp-CoffeeScript` that brings the latest features from [IJavascript](https://github.com/n-riesco/ijavascript).

This new release uses [zeromq](https://github.com/zeromq/zeromq.js), which fixes this issue by providing prebuilt ZMQ bindings for `Node.js`.

I've also added new commands:
- `jp-coffee-install` to install a kernel spec for `jp-CoffeeScript`
- `jp-coffee-console` to open a `jp-CoffeeScript` console
- ...

See the [README.md](https://github.com/n-riesco/jp-coffeescript) for more details.

Closed via https://github.com/n-riesco/jp-coffeescript/commit/b4ce06331b8276d7b33cd6253627a2d16293aa4c
